File: 11651/11651.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    11651.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/03/03 23:24:03 by jihoolee          #+#    #+#              #
#    Updated: 2022/03/03 23:37:22 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

length = int(input())
coords = [list(map(int, input().split())) for _ in range(length)]

sorted_coords = sorted(coords, key=lambda coord: (coord[1], coord[0]))
for coord in sorted_coords:
    print(coord[0], coord[1])

File: 11651/11651.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   11651.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/03/03 23:15:26 by jihoolee          #+#    #+#             */
/*   Updated: 2022/03/03 23:23:08 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>
#include <algorithm>

struct myclass {
  bool operator() (std::pair<int, int> lhs, std::pair<int, int> rhs) {
    if (lhs.second == rhs.second)
      return lhs.first < rhs.first;
    return lhs.second < rhs.second;
  }
} my_comp;

int main(void) {
  int len;
  std::cin >> len;
  std::vector<std::pair<int, int> >  vec(len);
  int x, y;

  for (int idx = 0; idx < len; ++idx) {
    std::cin >> x >> y;
    vec[idx] = std::make_pair(x, y);
  }
  std::sort(vec.begin(), vec.end(), my_comp);
  for (int idx = 0; idx < len; ++idx)
    std::cout << vec[idx].first << ' ' << vec[idx].second << '\n';
  return 0;
}
